### DISCUSSION TITLE
TxT2Vid - content blurring out after first frame and gets progressively worse

### DISCUSSION THREAD

**Discussion by TheShadowNYC** [volotat/SD-CN-Animation#85] [Show and tell]
First off, absolutely fantastic work Volotat! 

When generating via txt2vid I'm getting blurred results after the initial image. Attempted to try different models (inpainting included) as well as combos of different samplers/VAE/etc. Prompts are straight forward too and even when trying to add "in-focus, sharp, focused, etc" doesn't impact the result. Seems something else is going on or perhaps interacting with my setup somehow? Any thoughts? See samples below.

https://github.com/volotat/SD-CN-Animation/assets/116387255/ecfb41cd-7936-49a8-b1eb-6c7870db82e1

https://github.com/volotat/SD-CN-Animation/assets/116387255/9ce7dad4-70a8-4442-ad28-785757320822

**Comment by volotat** [comment]
This may happen with the fixed seed. Make sure it is set to -1 as it is not automatically resets by the next frame as it should.

**Reply by TheShadowNYC** [reply]
Thanks for your response. Yes, have been using -1 (default setting I believe). LMK if anything else to try out, happy to help in any way I can if issue is widespread among users.

**Reply by TheShadowNYC** [reply]
ran some more tests with varying settings and forcing -1 and then fixed seed (ie 15151515) with same results. Also tried to remove cmd argument (--no-half-vae) to see if changes anything. Fresh test below for reference. LMK if anything else I can try that might help trouble shoot.

https://github.com/volotat/SD-CN-Animation/assets/116387255/33d154fe-43ab-4831-99dd-aab23bb3ecfe

**Reply by volotat** [reply]
It seems that I found the source of the issue. It was related to how flow was applied to the image. Please try it out again. There may still be some blurring effect because of the motion itself, but it should not be that dramatic.

Here is a short test with new interpolation mode:

https://github.com/volotat/SD-CN-Animation/assets/14307880/ca81417c-80d2-4699-97c4-bce794772136

**Reply by TheShadowNYC** [reply]
Hi Volotat, thanks for looking into this further. Can't seem to replicate your result. I'm still seeing blurring occurring after frame one. Latest version throws a error now when CN is used btw... They released new version today with new "Reference Only" method. 

_"\controlnet.py", line 1063, in process
    raise ValueError('controlnet is enabled but no input image is given')
ValueError: controlnet is enabled but no input image is given"_

Could you be kind enough to confirm your model/VAE/ControNet setting used? I'm not changing anything else in your plugin settings btw.

Here are some results using same prompt but varying use of CN and models (RealisticVisionV20 and inpainting RVV20)

https://github.com/volotat/SD-CN-Animation/assets/116387255/db0ae835-12d1-4080-9f08-022cfbd29688


https://github.com/volotat/SD-CN-Animation/assets/116387255/c53c0f68-fe63-414e-b3ac-4e06740acd0e


https://github.com/volotat/SD-CN-Animation/assets/116387255/90690067-c445-4382-a4f9-cb12eafa2bbd

**Reply by TheShadowNYC** [reply]
posting test from latest version today... still seeing blurring after couple frames. Anything I should try changing in settings?

https://github.com/volotat/SD-CN-Animation/assets/116387255/ca913520-dfa0-48b9-a1a4-8f60bbb3843b


https://github.com/volotat/SD-CN-Animation/assets/116387255/9812e8cf-5e96-4ac1-82fb-c53bdb27f012

**Reply by volotat** [reply]
Could you share your parameters? You can get them through the 'export' button below.  And what model are you using? I need to reproduce this issue on my side to be able to fix it.

**Reply by TheShadowNYC** [reply]
Appreciate you looking into this!

Here is export.... and file it made. Using "RealisticVisionV20 safetensors" or "v1-5-pruned.safetensors" and Auto VAE or "vae-ft-mse-840000-ema-pruned" which gave same results.

https://github.com/volotat/SD-CN-Animation/assets/116387255/89714c2d-cfdb-43d8-a95e-950412d38df0



{
  "prompt": "close up of flower",
  "n_prompt": "text, letters, logo, brand, close up, cropped, out of frame, worst quality, low quality, jpeg artifacts, ugly, duplicate, morbid, mutilated, extra fingers, mutated hands, poorly drawn hands, poorly drawn face, mutation, deformed, blurry, dehydrated, bad anatomy, bad proportions, extra limbs, cloned face, disfigured, gross proportions, malformed limbs, missing arms, missing legs, extra arms, extra legs, fused fingers, too many fingers, long neck",
  "steps": 15,
  "sampler_index": 0,
  "cfg_scale": 5.5,
  "image_cfg_scale": 1.5,
  "fix_frame_strength": 0.35,
  "seed": -1,
  "height": 512,
  "width": 512,
  "fps": 12,
  "sdcn_process_mode": "txt2vid",
  "processing_strength": 0.85,
  "length": 10,
  "ControlNets": []
}

**Reply by TheShadowNYC** [reply]
also if helpful at all.... 

Python 3.10.0 (tags/v3.10.0:b494f59, Oct  4 2021, 19:00:18) [MSC v.1929 64 bit (AMD64)]
Version: v1.2.1
Commit hash: 89f9faa63388756314e8a1d96cf86bf5e0663045

 torch 1.13.1+cu117

**Comment by Lalimec** [comment]
how is this issue going?

**Reply by TheShadowNYC** [reply]
still same results, but saw a new commit today so will check again tomorrow!

**Reply by TheShadowNYC** [reply]
checked again... similar results... still blurs after first frame unfortunately. @volotat any updates on other settings to try and change on my end to fix?

BTW Volotat, shout out to you over in Yerevan! One of my best friends is based there (traditional 2D animator has his own studio).

**Reply by lauretta91** [reply]
Have you tried increasing the Fix Frame Strength ?

**Reply by TheShadowNYC** [reply]
oh wow, this works now! Have to find time to test it more. But mark this solved! Thanks

**Reply by jfranecki** [reply]
What did you do to fix it? Was Fix Frame Strength too low?

**Reply by TheShadowNYC** [reply]
yes, that was new setting in recent version it seems, I got fixed results after increasing to 7-8... too high and it became erratic movements but not blurry still. Adding Control Net source really helps too! Can't wait to play with it more.
